holidays/countries/guyana.py

```diff
@@ -0,0 +1,263 @@
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+):
+    """Guyana holidays.
+
+    References:
+        * [Public Holidays Amendment Act 1967](https://web.archive.org/web/20241125073357/https://parliament.gov.gy/documents/bills/11264-bill_6_of_1967_public_holidays.pdf)
+        * [Public Holidays Amendment Act 1969](https://web.archive.org/web/20250424211718/https://parliament.gov.gy/documents/bills/11891-bill_26_of_1969_1.pdf)
+        * [Public Holidays Act Consolidated as of 2012](https://mola.gov.gy/laws/Volume%206%20Cap.%2018.01%20-%2023.011696968337.pdf)
+    """
+
+    country = "GY"
+    default_language = "en_GY"
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("%s (observed, estimated)")
+    start_year = 1968
+    supported_languages = ("en_GY", "en_US")
+
+    def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=GuyanaHinduHolidays)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=GuyanaIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        StaticHolidays.__init__(self, cls=GuyanaStaticHolidays)
+        kwargs.setdefault("observed_rule", SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(
+            self._add_new_years_day(tr("New Year's Day")), rule=SAT_SUN_TO_NEXT_WORKDAY
+        )
+
+        if self._year <= 1969:
+            # Independence Day.
+            self._add_holiday_may_26(tr("Independence Day"))
+
+        if self._year >= 1970:
+            # Republic Day.
+            self._add_holiday_feb_23(tr("Republic Day"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        # Labor Day.
+        self._add_observed(self._add_labor_day(tr("Labour Day")))
+
+        # Commonwealth Day.
+        self._add_holiday_1st_mon_of_aug(tr("Commonwealth Day"))
+
+        # Christmas Day.
+        self._add_observed(
+            self._add_christmas_day(tr("Christmas Day")), rule=SAT_SUN_TO_NEXT_MON_TUE
+        )
+
+        # Day after Christmas.
+        self._add_observed(
+            self._add_christmas_day_two(tr("Day after Christmas")), rule=SAT_SUN_TO_NEXT_MON_TUE
+        )
+
+        # Phagwah Day.
+        self._add_observed(self._add_holi(tr("Phagwah Day")))
+
+        # Deepavali Day.
+        self._add_observed(self._add_diwali(tr("Deepavali Day")))
```

Comment:
```suggestion
        name = tr("Deepavali Day")
        self._add_observed(
            self._add_diwali(name) if self._year >= 2016 else self._add_holiday_oct_30(name)
        )
```
If this holiday really had a fixed date before 2016, then it is better to make it that way.

Reply:
The source of the dates is https://www.timeanddate.com/holidays/guyana/deepavali, it feels incorrect to me though but I don't have conclusive evidence for either? Should I update this to just use `_add_diwali` always?

Reply:
Yes, it looks like an error in Timeanddate data. There is no mention of fixed date anywhere. So it's reasonable to make it like regular Diwali holiday.

Reply:
I see that the dates for Diwali in Guyana align with DIWALI_INDIA_DATES except for 2016 and 2018. Should I keep the custom dates for 2016 to 2025 and for years before 2016 use DIWALI_INDIA_DATES? Or should we just support Diwali for 2016-onwards for Guyana?

Reply:
Yes, use `_add_diwali_india`, and DIWALI_INDIA_DATES for 2016-2025.